Bug: helper/resource/testing.modulePathPrimaryInstanceState panic on wrong path

When providing a non-existent path to a test helper (e.g. resource.TestCheckModuleResourceAttrSet), terraform panics. It should instead raise an error.

PR to follow.

```
=== RUN   TestAccArrakisScheduledTask
--- FAIL: TestAccArrakisScheduledTask (20.38s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xc0d157]

goroutine 6 [running]:
testing.tRunner.func1(0xc4203feb40)
	/usr/local/go/src/testing/testing.go:711 +0x2d2
panic(0x2359d80, 0x3f08760)
	/usr/local/go/src/runtime/panic.go:491 +0x283
app/vendor/github.com/hashicorp/terraform/helper/resource.modulePrimaryInstanceState(0xc42069ecc0, 0x0, 0x286f3cd, 0x1c, 0x0, 0xc420ae9ba0, 0x0)
	/go/src/app/vendor/github.com/hashicorp/terraform/helper/resource/testing.go:1121 +0x37
app/vendor/github.com/hashicorp/terraform/helper/resource.modulePathPrimaryInstanceState(0xc42069ecc0, 0x3f0b240, 0x2, 0x2, 0x286f3cd, 0x1c, 0x42cbe6, 0x28cac30, 0xc420697070)
	/go/src/app/vendor/github.com/hashicorp/terraform/helper/resource/testing.go:1138 +0x79
app/vendor/github.com/hashicorp/terraform/helper/resource.TestCheckModuleResourceAttrSet.func1(0xc42069ecc0, 0xa900df, 0xc4208d8660)
	/go/src/app/vendor/github.com/hashicorp/terraform/helper/resource/testing.go:856 +0x78
app/vendor/github.com/hashicorp/terraform/helper/resource.ComposeTestCheckFunc.func1(0xc42069ecc0, 0xc42069ecc0, 0x0)
	/go/src/app/vendor/github.com/hashicorp/terraform/helper/resource/testing.go:806 +0x88
app/vendor/github.com/hashicorp/terraform/helper/resource.testStep(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/app/vendor/github.com/hashicorp/terraform/helper/resource/testing_config.go:81 +0xddc
app/vendor/github.com/hashicorp/terraform/helper/resource.testStepConfig(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/app/vendor/github.com/hashicorp/terraform/helper/resource/testing_config.go:17 +0x98
app/vendor/github.com/hashicorp/terraform/helper/resource.Test(0x3ea43e0, 0xc4203feb40, 0x0, 0xc420203c70, 0xc4203fc0f0, 0x0, 0x0, 0x0, 0xc4201a4420, 0x1, ...)
	/go/src/app/vendor/github.com/hashicorp/terraform/helper/resource/testing.go:487 +0x142f
app/tests.TestAccArrakisScheduledTask(0xc4203feb40)
	/go/src/app/tests/arrakis_scheduled_task_test.go:19 +0x260
testing.tRunner(0xc4203feb40, 0x28c5e90)
	/usr/local/go/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:789 +0x2de
exit status 2
FAIL	app/tests	20.406s
```

--- a/helper/resource/testing.go
+++ b/helper/resource/testing.go
@@ -1135,6 +1135,10 @@ func modulePrimaryInstanceState(s *terraform.State, ms *terraform.ModuleState, n
 // given resource name in a given module path.
 func modulePathPrimaryInstanceState(s *terraform.State, mp []string, name string) (*terraform.InstanceState, error) {
 	ms := s.ModuleByPath(mp)
+	if ms == nil {
+		return nil, fmt.Errorf("No module found at: %s", mp)
+	}
+
 	return modulePrimaryInstanceState(s, ms, name)
 }
 

